### DISCUSSION TITLE
Specification version 4

### DISCUSSION THREAD

**Discussion by KonradLinkowski** [pact-foundation/pact-js#871] [Q&A] [answered]
Is there any way to use pact specification in version 4 in JavaScript?
Seems like the other libraries already use v4 and this one still uses v2 and v3 in the beta version.

**Comment by YOU54F** [comment, marked as answer]
Hey V4 spec is available in JVM and Rust at the moment,



taken from https://pactflow.io/blog/changelog-oss-may-2022/

V4 Pact Specification

> Pact specification V4 is here! We worked hard on listening to the community, after the release of v3, 5 years ago. We consolidation many of the requests people made about things that haven't worked quite properly. One of the biggest changes is consolidation the file format to allow for HTTP and message interactions with a single file.
> :video_camera: See an AMA from 2021 where Ron Holshausen took us through "What's in store for Pact specification v4.0?"
> :orange_book: Read the V4 spec RFC for a full list of changes: https://github.com/pact-foundation/pact-specification/issues/71
> :rocket: Implementations in Rust core and JVM has been completed
> :earth_africa: All our Pact specifications are open-source and you can see them all here
> Plugins, Protobufs and gRPC (oh my!)


I'm not sure if we have tracking tickets for the v3/v4 in each language, but I hope to update and reboot our feature table soon https://docs.pact.io/roadmap/feature_support and get some tickets tracking work, if we don't already have them either in each repo or on our canny board https://pact.canny.io/ where you can submit a feature request, and people can vote on it
